lib/app/features/user/providers/avatar_picker_notifier.dart

```diff
@@ -15,44 +16,49 @@ sealed class AvatarPickerState with _$AvatarPickerState {
 
   const factory AvatarPickerState.picked({required MediaFile file}) = AvatarPickerStatePicked;
 
-  const factory AvatarPickerState.compressed({required MediaFile file}) =
-      AvatarPickerStateCompressed;
+  const factory AvatarPickerState.cropped({required MediaFile file}) = AvatarPickerStateCropped;
+
+  const factory AvatarPickerState.processed({required MediaFile file}) = AvatarPickerStateProcessed;
 
   const factory AvatarPickerState.error({required String message}) = AvatarPickerStateError;
 }
 
-@Riverpod()
+@riverpod
 class AvatarPickerNotifier extends _$AvatarPickerNotifier {
   @override
   AvatarPickerState build() {
     return const AvatarPickerState.initial();
   }
 
-  Future<void> pick({
+  Future<void> process({
+    required String assetId,
     required CropImageUiSettings cropUiSettings,
-    required Future<MediaFile?> Function() pickMediaFile,
   }) async {
     final mediaService = ref.read(mediaServiceProvider);
     final compressService = ref.read(mediaCompressServiceProvider);
 
     try {
-      final mediaFile = await pickMediaFile();
-      if (mediaFile == null) return;
-      final cameraImagePath = await ref.read(assetFilePathProvider(mediaFile.path).future);
-      if (cameraImagePath == null) return;
+      final assetImagePath = await ref.read(assetFilePathProvider(assetId).future);
+      if (assetImagePath == null) {
+        throw AssetEntityFileNotFoundException();
+      }
+      state = AvatarPickerState.picked(file: MediaFile(path: assetImagePath));
       final croppedImage = await mediaService.cropImage(
         uiSettings: cropUiSettings,
-        path: cameraImagePath,
+        path: assetImagePath,
       );
-      if (croppedImage == null) return;
-      state = AvatarPickerState.picked(file: croppedImage);
+      if (croppedImage == null) {
+        state = const AvatarPickerState.initial();
```

Comment:
Maybe we can show some messages to the user? 
If cropping or compression fails, the state reverts to initial, but the user is not informed, potentially leading to confusion.